packages/auth/src/clients/service.test.ts

```diff
@@ -0,0 +1,332 @@
+            }
+          }
+        })
+
+        expect(validClient).toEqual(true)
+      })
+
+      test('Cannot validate client with incorrect display name', async (): Promise<void> => {
+        const expDate = new Date()
+        expDate.setTime(expDate.getTime() + 1000 * 60 * 60)
+
+        const nbfDate = new Date()
+        nbfDate.setTime(nbfDate.getTime() - 1000 * 60 * 60)
+        nock(KEY_REGISTRY_ORIGIN)
+          .get(TEST_KID_PATH)
+          .reply(200, {
+            display: {
+              name: 'Wrong Client',
+              uri: 'https://example.com'
+            },
+            keys: {
+              ...TEST_CLIENT_KEY,
+              exp: Math.round(expDate.getTime() / 1000),
+              nbf: Math.round(nbfDate.getTime() / 1000),
+              revoked: false
+            }
+          })
+
+        const validClient = await clientService.validateClientWithRegistry({
+          display: TEST_CLIENT_DISPLAY,
+          key: {
+            proof: 'httpsig',
+            jwk: TEST_CLIENT_KEY
+          }
+        })
+
+        expect(validClient).toEqual(false)
+      })
+
+      test('Cannot validate client with incorrect uri', async (): Promise<void> => {
+        const expDate = new Date()
+        expDate.setTime(expDate.getTime() + 1000 * 60 * 60)
+        const nbfDate = new Date()
+        nbfDate.setTime(nbfDate.getTime() - 1000 * 60 * 60)
+        nock(KEY_REGISTRY_ORIGIN)
+          .get(TEST_KID_PATH)
+          .reply(200, {
+            display: {
+              name: 'Test Client',
+              uri: 'https://example.com/wrong'
+            },
+            keys: {
+              ...TEST_CLIENT_KEY,
+              exp: Math.round(expDate.getTime() / 1000),
+              nbf: Math.round(nbfDate.getTime() / 1000),
+              revoked: false
+            }
+          })
+
+        const validClient = await clientService.validateClientWithRegistry({
+          display: TEST_CLIENT_DISPLAY,
+          key: {
+            proof: 'httpsig',
+            jwk: TEST_CLIENT_KEY
+          }
+        })
+
+        expect(validClient).toEqual(false)
+      })
+    })
+
+    test('Cannot validate client with key that has invalid properties', async (): Promise<void> => {
+      const futureDate = new Date()
+      futureDate.setTime(futureDate.getTime() + 1000 * 60 * 60)
+      const nbfDate = new Date()
+      nbfDate.setTime(nbfDate.getTime() - 1000 * 60 * 60)
+      nock(KEY_REGISTRY_ORIGIN)
+        .get(TEST_KID_PATH)
+        .reply(200, {
+          display: TEST_CLIENT_DISPLAY,
+          keys: {
+            ...TEST_CLIENT_KEY,
+            exp: Math.round(futureDate.getTime() / 1000),
+            nbf: Math.round(nbfDate.getTime() / 1000),
+            revoked: false
+          }
+        })
+
+      nock(KEY_REGISTRY_ORIGIN).get('/wrong').reply(200)
+
+      // Validate kid
+      const validClientKid = await clientService.validateClientWithRegistry({
```

Comment:
Is this intended to test (which would require using a different origin)?
https://github.com/interledger/rafiki/pull/328/files#diff-ebe1c87507722350de717578d84973dc69709f084f1195e4221dfaf97bcdd9daR111
It returns false at
https://github.com/interledger/rafiki/pull/328/files#diff-ebe1c87507722350de717578d84973dc69709f084f1195e4221dfaf97bcdd9daR127

Reply:
It's supposed to test when a `kid` is a url but doesn't point to an endpoint with a key or is otherwise invalid.